### PR TITLE
feat: health check api (watcher)

### PR DESCRIPTION
**Pull request by adrysn**
This PR adds support agent host health check API by watcher daemon.

Internal Ticket: OP#664
Related: https://github.com/lablup/backend.ai-manager/pull/351, https://github.com/lablup/backend.ai-common/pull/60

### CODE REVIEW

**Review by achimnol — CHANGES_REQUESTED** [review]
nvidia-smi supports CSV-formatted outputs for programmatic consumption.
Please consider using that output:
```
> nvidia-smi --format=csv --query-gpu=index,name,uuid,driver_version,memory.total,memory.used,utilization.gpu,utilization.memory --no-units
index, name, uuid, driver_version, memory.total [MiB], memory.used [MiB], utilization.gpu [%], utilization.memory [%]
0, Tesla V100-SXM3-32GB, GPU-02ba5a4d-2829-ebf7-e613-083e8b62dbb5, 418.126.02, 32480 MiB, 13 MiB, 0 %, 0 %
1, Tesla V100-SXM3-32GB, GPU-385c62a7-0a95-cce5-81a8-2c9c5d5a7f6c, 418.126.02, 32480 MiB, 13 MiB, 0 %, 0 %
2, Tesla V100-SXM3-32GB, GPU-3c6b25e5-f7b9-ffc9-2c3c-386586c0778b, 418.126.02, 32480 MiB, 13 MiB, 0 %, 0 %
3, Tesla V100-SXM3-32GB, GPU-0e89496d-bcbf-b990-ed48-3859751f5db2, 418.126.02, 32480 MiB, 13 MiB, 0 %, 0 %
4, Tesla V100-SXM3-32GB, GPU-331cc9a6-dfc2-3b04-c8b0-f004d14e0483, 418.126.02, 32480 MiB, 13 MiB, 0 %, 0 %
5, Tesla V100-SXM3-32GB, GPU-3cd84123-d9ff-6032-1bc7-ab191f7a980c, 418.126.02, 32480 MiB, 13 MiB, 0 %, 0 %
6, Tesla V100-SXM3-32GB, GPU-361a9076-6bcc-3875-771b-411116e1ce63, 418.126.02, 32480 MiB, 13 MiB, 0 %, 0 %
7, Tesla V100-SXM3-32GB, GPU-fdd89a92-8b04-f092-3e07-b434249a0542, 418.126.02, 32480 MiB, 13 MiB, 0 %, 0 %
8, Tesla V100-SXM3-32GB, GPU-d929421b-3377-39c0-b9b2-474ee68afb10, 418.126.02, 32480 MiB, 13 MiB, 0 %, 0 %
9, Tesla V100-SXM3-32GB, GPU-3caa4672-eb42-1b6a-9857-d520fb715484, 418.126.02, 32480 MiB, 13 MiB, 0 %, 0 %
10, Tesla V100-SXM3-32GB, GPU-529aa577-f39c-ca7f-5162-ce888a2fecbc, 418.126.02, 32480 MiB, 13 MiB, 0 %, 0 %
11, Tesla V100-SXM3-32GB, GPU-5318f22e-03eb-0ae6-6b8f-f7be360fb3c2, 418.126.02, 32480 MiB, 13 MiB, 0 %, 0 %
12, Tesla V100-SXM3-32GB, GPU-1ca51595-f08d-c2f2-1655-7e53908d7730, 418.126.02, 32480 MiB, 13 MiB, 0 %, 0 %
13, Tesla V100-SXM3-32GB, GPU-3a0319d2-a635-eb2e-b105-32c38916f300, 418.126.02, 32480 MiB, 13 MiB, 0 %, 0 %
14, Tesla V100-SXM3-32GB, GPU-6f3d0fce-31b4-fc44-df3d-ca81a88aea72, 418.126.02, 32480 MiB, 13 MiB, 0 %, 0 %
15, Tesla V100-SXM3-32GB, GPU-bf934fea-6ded-5377-5804-672911facb2b, 418.126.02, 32480 MiB, 13 MiB, 0 %, 0 %
```

**Comment by adrysn** [comment]
Updated to return GPU detail information if found.

File: src/ai/backend/agent/watcher.py [hunk 1]
```diff
@@ -237,33 +238,48 @@ async def health_check(request: web.Request) -> web.Response:
...
     elif err:
         nvidia_info = {
             'status': 'error',
             'message': 'check NVIDIA GPU status',
         }
-    elif 'NVIDIA-SMI' in out:
+
+    gpu_details = []
+    for gpu_info in out.split('\n')[1:]:
+        gpu_items = gpu_info.split(', ')
```
Comment (achimnol) [inline src/ai/backend/agent/watcher.py:262]: Python has an intrinsic CSV parser, which handles a few more edge cases such as quotes and escapes.
How about using it?
ref) https://docs.python.org/3/library/csv.html

Comment (adrysn) [reply src/ai/backend/agent/watcher.py]: Replaced the csv parsing logic to use csv module in the standard library.

**Comment by adrysn** [comment]
This is deprecated in favor of https://github.com/lablup/backend.ai-watcher.

**Pull request closed without merging**